soh/soh/Enhancements/mods.cpp

```diff
@@ -486,6 +486,21 @@ void RegisterBonkDamage() {
     });
 }
 
+void RegisterPathFix() {
+    GameInteractor::Instance->RegisterGameHook<GameInteractor::OnTransitionEnd>([](int32_t sceneNum) {
+        switch (sceneNum) {
+            case SCENE_SPOT00:
+            case SCENE_SPOT04:
+            case SCENE_SPOT15:
+                CVarSetInteger("gDirtPathFix", CVarGetInteger("gSceneSpecificDirtPathFix", 0));
```

Comment:
i'm not the biggest fan of `gSceneSpecificDirtPathFix` as a name here, i think it'd be more clear as `gMenuOptionDirtPathFix`

that way it's communicating that in these specific scenes, we apply the dirt path fix based on the option for the menu, but in all other scenes we ignore it

Reply:
Sounds good to me. I wasn't sure about the name so I just went with that.